controllers/orderController.js

```diff
@@ -26,6 +26,7 @@ const createOrder = async (req, res) => {
 
   let orderItems = []
   let subtotal = 0
+  // suggest reduce
   // Iterating over the cart items
   for (const item of cartItems) {
```

Comment:
What you have with the `for...of` loop totally works. Just wanted to share another way you can write this code using the `reduce` iterator function:

```suggestion
  const {orderItems, subtotal} = cartItems.reduce(async (cartItem, result) => {
    // Finding the product in the database based on the item's product ID
    const dbProduct = await Product.findOne({ _id: item.product })

    if (!dbProduct) {
      throw new CustomError.NotFoundError(
        `No product with id : ${item.product}`
      )
    }

    const { name, price, image, _id } = dbProduct
    // Creating an object for each item in the order
    const singleOrderItem = {
      amount: item.amount,
      name,
      price,
      image,
      product: _id
    }

    const result = await result;
    // Adding the item to the order
    result.orderItems = [...result.orderItems, singleOrderItem]
    // Calculating the order subtotal
    result.subtotal += item.amount * price
    return result;
  }, { orderItems: [], subtotal: 0 })
```

Reply:
Thank you a lot for your suggestion🔥